toolchain/check/handle_function.cpp

```diff
@@ -333,6 +333,32 @@ static auto RequestVtableIfVirtual(
     CARBON_DIAGNOSTIC(ImplWithoutBase, Error, "impl without base class");
     context.emitter().Emit(node_id, ImplWithoutBase);
   }
+
+  if (generic_id != SemIR::GenericId::None) {
+    bool function_is_generic = true;
+    if (class_info.generic_id != SemIR::GenericId::None) {
+      const auto& function_generic = context.generics().Get(generic_id);
+      const auto& class_generic = context.generics().Get(class_info.generic_id);
+
+      auto function_bindings =
+          context.inst_blocks().Get(function_generic.bindings_id);
+      auto class_bindings =
+          context.inst_blocks().Get(class_generic.bindings_id);
+
+      // If the function's bindings are the same size as the class's bindings,
+      // then there are no extra bindings for the function, so it is effectively
+      // non-generic within the scope of a specific of the class.
+      if (class_bindings.size() == function_bindings.size()) {
+        function_is_generic = false;
+      }
+    }
```

Comment:
My sad comment is, maybe make this a `IsGenericFunction` helper to allow for early returns rather than nested conditions.

Reply:
Oh, sure!  3c61542